packages/vscode-extension/src/service/ImportObjectService.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import SuiteCloudRunner from '../core/SuiteCloudRunner';
+import DummyConsoleLogger from '../loggers/DummyConsoleLogger';
+import VSConsoleLogger from '../loggers/VSConsoleLogger';
+import { ConsoleLogger } from '../util/ExtensionUtil';
+import MessageService from './MessageService';
+import { IMPORT_OBJECTS, LIST_OBJECTS } from './TranslationKeys';
+import { VSTranslationService } from './VSTranslationService';
+
+const IMPORT_OBJECT_COMMAND_NAME = 'object:import';
+const LIST_OBJECT_COMMAND_NAME = 'object:list';
+const CONSOLE_LOGGER_ERROR = 'vsConsole Logger not initialized';
+
+export default class CustomObjectService {
+	private executionPath?: string;
+	private readonly messageService: MessageService;
+	private vsConsoleLogger: VSConsoleLogger | undefined;
+	private translationService: VSTranslationService;
+
+	constructor(messageService: MessageService, translationService: VSTranslationService) {
+		this.messageService = messageService;
+		this.translationService = translationService;
+	}
+
+	async importObjects(
+		destinationFolder: string,
+		appId: string | undefined,
+		scriptIds: string[] | undefined,
+		includeReferencedFiles: boolean,
+		executionPath: string | undefined
+	) {
+		this.executionPath = executionPath;
+		//We choose 'ALL' types because it is chosen before which scriptIds should be imported and not by type
+		let commandArgs: any = { type: 'ALL', destinationfolder: destinationFolder };
+
+		commandArgs.scriptid = scriptIds && scriptIds.length > 0 ? scriptIds : 'ALL';
+
+		if (appId === undefined || appId.length != 0) {
+			commandArgs.appid = appId;
+		}
+
+		if (!includeReferencedFiles) {
+			commandArgs.excludefiles = true;
+		}
+
+		const commandActionPromise = this.runSuiteCloudCommand(commandArgs, IMPORT_OBJECT_COMMAND_NAME);
+		const statusBarMessage = this.translationService.getMessage(IMPORT_OBJECTS.IMPORTING_OBJECTS);
+		this.messageService.showStatusBarMessage(statusBarMessage, true, commandActionPromise);
+		return await commandActionPromise;
+	}
+
+	async listObjects(appId: string | undefined, types: string[], scriptId: string | undefined, executionPath: string | undefined) {
```

Comment:
This actually the signature that I would like to change, to not accept undefined values. I think it will make our code more robust, because the typescript check would avoid undesired calls with parameters unset. Instead we will be at least forced to pass emptyStrings which are more difficult to pass by accident.

Reply:
Again, I don't see any problem in having it undefined. Line 59 is waiting on that. Same for scriptid. 
Only one I see it's worthy to control is executionPath that is already checked in validate()